.travis.yml

```diff
@@ -1,7 +1,230 @@
 sudo: false
-install: git clone --depth 1 https://github.com/sstephenson/bats.git
-script: make test
 language: c
+
+addons:
+  apt:
+    packages:
+      - python-virtualenv
+
 env:
-  - PYENV_NATIVE_EXT=
-  - PYENV_NATIVE_EXT=1
+  global:
+    - VIRTUAL_ENV_BASE=~/virtualenv
+
+matrix:
+  include:
+    # Plain bats
+    - env: PYENV_NATIVE_EXT=
+      addons: false
+    - env: PYENV_NATIVE_EXT=1
+      addons: false
+
+    # PyPy
+    - env: TESTENV=pypy-2.4.0 DIST=precise
+      dist: precise
+
+    - env: TESTENV=pypy-5.1.1 DIST=precise
+      dist: precise
+
+    - env: TESTENV=pypy-5.1.1 DIST=trusty
+      dist: trusty
+
+    # Miniconda
+
+    - env: TESTENV=miniconda3-4.0.5 DIST=precise LD_LIBRARY_PATH=/opt/python/3.5.0/lib/
```

Comment:
The need for LD_LIBRARY_PATH here seems like it could be fixed by pyenv, but also fixed upstream...?